src/pages/Home.tsx

```
@@ -0,0 +1,76 @@
+import { Button } from 'primereact/button'
+import { useSelector, useDispatch } from 'react-redux'
+import type { RootState, AppDispatch } from '../store'
+import { useState } from 'react'
+import { increment, decrement } from '../store'
+import { Avatar } from 'primereact/avatar'
+import axios from 'axios'
+import { Dropdown } from 'primereact/dropdown'
+import { setUser } from '../features/marketplace/slices/userSlice'
+import Cookies from 'js-cookie'
+import UpdateRole from '../components/shared/UpdateRole'
+import type { RoleType, User } from '../types/index';
+
+const roleOptions: { label: string; value: RoleType }[] = [
+  { label: 'Customer', value: 'customer' },
+  { label: 'Manufacturer', value: 'manufacturer' },
+  { label: 'Creator', value: 'creator' },
+  { label: 'Admin', value: 'admin' }, // אם צריך גם admin
+];
+
+function Home() {
+  const [count, setCount] = useState(0);
+  const counter = useSelector((state: RootState) => state.counter.value);
+  const dispatch = useDispatch<AppDispatch>();
+
+  const user = useSelector((state: RootState) => state.user);
+  const [showRoleUpdate, setShowRoleUpdate] = useState(false);
+  
+  const getInitialRole = (role: string | null | undefined): RoleType => {
+    const allowed: RoleType[] = ['customer', 'manufacturer', 'creator', 'admin'];
+    return allowed.includes(role as RoleType) ? (role as RoleType) : 'customer';
```

Consider validating `role` more safely instead of using `as` casting – it may silently accept invalid strings.